site/_docs/core-java.md

```diff
@@ -233,14 +233,15 @@ VectorizedRowBatch batch = schema.createRowBatch();
 LongColumnVector x = (LongColumnVector) batch.cols[0];
 LongColumnVector y = (LongColumnVector) batch.cols[1];
 for(int r=0; r < 10000; ++r) {
-  int row = batch.size++;
+  int row = batch.size;
   x.vector[row] = r;
   y.vector[row] = r * 3;
   // If the batch is full, write it out and start over.
   if (batch.size == batch.getMaxSize()) {
     writer.addRowBatch(batch);
     batch.reset();
   }
+  batch.size++;
 }
 writer.close();
```

Comment:
That's correct. I already fixed it in my code. Had also this issue when the batch smaller than the max size.